### PR TITLE
feat: Save cart UI #9191

### LINKED ISSUE #9191: Save cart UI
Create UI for the "Save cart" button as well as the "Saved Carts" action link to visit the saved carts listing page.

Create one component for both buttons, this component is only for b2b and should be CMS driven. 

### Scope
- Create component in Spartacus sample data
- Create Spartacus component matching UI to render the CMS component
- Create a dialog to enter name and description that can be reused for the edit feature
- Hook up the save cart logic from #9190 to the button
- The saved cart link will be wired later

### Acceptance
- Should be able to click on the saved cart button to save cart

### PR DESCRIPTION
**Pull request by dydome**
close #9191

### CODE REVIEW

File: feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  templateUrl: './add-to-saved-cart.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AddToSavedCartComponent implements OnInit, OnDestroy {
+  cart$: Observable<Cart>;
+  loggedIn = false;
+
+  @ViewChild('element') element: ElementRef;
+
+  private subscription = new Subscription();
```
Comment (mszczepaniak) [inline feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.ts]:
```suggestion
  protected subscription = new Subscription();
```

File: feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+      if (dialog) {
+        this.subscription.add(dialog.pipe(take(1)).subscribe());
+      }
+    } else {
+      this.routingService.go({ cxRoute: 'login' });
+    }
+  }
+
+  private openDialog(
```
Comment (mszczepaniak) [inline feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.ts]:
```suggestion
  protected openDialog(
```

File: feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      vcr,
+      data
+    );
+
+    if (component) {
+      return combineLatest([
+        component,
+        this.launchDialogService.dialogClose,
+      ]).pipe(
+        filter(([, close]) => close && close !== undefined),
```
Comment (mszczepaniak) [inline feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.ts]:
```suggestion
        filter(([, close]) => close !== undefined),
```

File: feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import { filter, map, take, tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-add-to-saved-cart',
+  templateUrl: './add-to-saved-cart.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AddToSavedCartComponent implements OnInit, OnDestroy {
+  cart$: Observable<Cart>;
+  loggedIn = false;
```
Comment (mszczepaniak) [inline feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.ts]:
```suggestion
  protected loggedIn = false;
```

File: feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.html [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      [routerLink]="
+        {
+          cxRoute: 'savedCarts'
+        } | cxUrl
+      "
+      cxAutoFocus
+    >
+      {{ 'addToSavedCart.savedCarts' | cxTranslate }}
+    </button>
+    |
```
Comment (mszczepaniak) [inline feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.html]:
```suggestion
    
```
leftover?

File: feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+  constructor(
+    protected activeCartService: ActiveCartService,
+    protected authService: AuthService,
+    protected launchDialogService: LaunchDialogService,
+    protected routingService: RoutingService,
+    protected vcr: ViewContainerRef
+  ) {}
+
+  ngOnInit() {
```
Comment (mszczepaniak) [inline feature-libs/cart/saved-cart/components/add-to-saved-cart/add-to-saved-cart.component.ts]:
```suggestion
  ngOnInit(): void {
```

File: feature-libs/cart/saved-cart/components/saved-cart-form/saved-cart-form.component.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+  saveCart(): void {
+    // TO DO
+  }
+
+  dismissModal(): void {
+    this.close('Cancel click');
+  }
+
+  protected build() {
```
Comment (mszczepaniak) [inline feature-libs/cart/saved-cart/components/saved-cart-form/saved-cart-form.component.ts]:
```suggestion
  protected build(): void {
```

File: feature-libs/cart/saved-cart/components/saved-cart-form/saved-cart-form.component.ts [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+  get descriptionsCharacterLeft(): number {
+    return (
+      this.descriptionMaxLength -
+      (this.form.get('description')?.value?.length || 0)
+    );
+  }
+
+  saveCart(): void {
+    // TO DO
```
Comment (mszczepaniak) [inline feature-libs/cart/saved-cart/components/saved-cart-form/saved-cart-form.component.ts:35]: To do later or leftover?

File: projects/storefrontapp/src/environments/environment.ts [hunk 9]
```diff
@@ -15,7 +15,9 @@ import { Environment } from './models/environment.model';
```
Comment (mszczepaniak) [inline projects/storefrontapp/src/environments/environment.ts:15]: I do not think that you should commit the change of environment, it has a risk of later forgetting about removing that change

File: feature-libs/cart/saved-cart/components/saved-cart-form/saved-cart-form.component.ts (hunk 8 above)
Comment (dydome) [reply feature-libs/cart/saved-cart/components/saved-cart-form/saved-cart-form.component.ts:35]: This PR is only for UI part, this method is ready for state and facade PR.

File: projects/storefrontapp/src/environments/environment.ts (hunk 9 above)
Comment (dydome) [reply projects/storefrontapp/src/environments/environment.ts:15]: We are using this env in epic. So in the end we will change to develop variables.

**Review by mszczepaniak — APPROVED** [review]

**Pull request merged**